dashboard-frontend/src/types/study.ts

```diff
@@ -0,0 +1,33 @@
+export interface Study {
+  id: string;
+  name: string;
+  slug: string;
+  createdAt: string;
+  progress: StudyProgress;
+  strata: Strata[];
+}
+
+interface StudyProgress {
+  currentParticipants: number;
+  expectedParticipants: number;
+  currentAverageDeviation: number;
+  expectedAverageDeviation: number;
+  date: string;
+  averageDeviationOverTime: DataPoint[];
+}
+
+interface Strata {
+  id: string;
+  percentageDesired: number;
+  currentPercentage: number;
```

Comment:
Ah, in addition to the `percentage` numbers, we need the `absolute` numbers too. 

The former could be computed from the latter on the client-side, of course. But we need it everywhere, so maybe it's better to just keep shuffling both types around everywhere.

Reply:
Makes sense!

From having:
```
percentageDesired: number;
currentPercentage: number;
```

Now, we have:
```
desiredPercentage: number;
currentPercentage: number;
desiredValue: number;
currentValue: number;
```

I agree on what you say about not having to compute some things on the frontend. I added a **name field** to the **StratumProgress** interface.

I think that is good idea to treat id as an opaque token independently that right now the id is actually the name, what do you think on having this extra **name field** ?